bundles/org.openhab.core.io.transport.mqtt/src/main/java/org/eclipse/smarthome/io/transport/mqtt/MqttBrokerConnection.java

```diff
@@ -890,7 +888,7 @@ public void publish(String topic, byte[] payload, MqttActionCallback listener) {
      *         exceptionally on an error or with a result of false if no broker connection is established.
      */
     public CompletableFuture<Boolean> publish(String topic, byte[] payload) {
```

Comment:
Mark this deprecated maybe? Implicitly it uses deprecated class fields ("retain", "qos").

Reply:
It'll use the default (QoS 0 and retain false) after the getter/setter is removed.

Reply:
While we are discussing deprecation: What is the use-case for `subscribeRaw` and `unsubscribeRaw`? If there are no subscribers, why should the connection subscribe to a topic?

Reply:
I think it had to do keep alive subscriptions. But not sure anymore, I should have documented those methods better. Did you encounter any usages?

Reply:
Not on openhab2-addons and not in openhab-core

Reply:
Deprecating or better removing should be fine then. I also do not see any benefit in keeping this API.